ADR/ADRYYY-store-allowed-domain-list-as-file-in-forms-deploy.md

```diff
@@ -0,0 +1,126 @@
+flowchart TD
+    subgraph forms-admin
+    start([User visits GOV.UK Forms])
+    --> Auth0
+
+    subgraph Auth0
+    login_id_prompt[Ask user for email address]
+    -->
+
+    restrict_users_to_allowed_domains{Is the email address domain<br>in the allow list?}
+    -- No -->
+    unauthorised_email_domain([Show error<br>'Enter a government email address'])
+
+    restrict_users_to_allowed_domains -- Yes -->
+    authenticate_email_address[[Check that the user has<br>access to that email inbox]]
+    end
+
+    authenticate_email_address -->
+    find_or_create_user{Has the user signed in before?}
+
+    find_or_create_user -- No -->
+    create_user[Create a new user record]
+    -->
+    deny_users_with_restricted_domains{Is the email address domain<br>in the restricted list?}
+
+    deny_users_with_restricted_domains -- No -->
+    set_has_access_true[Allow access for the new user]
+    --> has_access
+
+    deny_users_with_restricted_domains -- Yes -->
+    set_has_access_false[Deny access for the new user]
+    --> has_access
+
+    find_or_create_user -- Yes -->
+    has_access{Does the user have access?}
+
+    has_access -- No -->
+    access_denied([Show error 'You do not have permission'])
+    has_access -- Yes -->
+    govuk_forms_home_page([Welcome to GOV.UK Forms])
+    end
+
+```
+
+The identity provider we are using currently is Auth0; it is configured using Terraform code in the `forms-deploy` repository. The `forms-admin` app is built and run as a container. The container image is built from code in the `forms-admin` repository and deployed as part of a release pipeline. This pipeline is defined using code in the `forms-deploy` repo, and also itself retrieves the `forms-deploy` repo to determine how the `forms-admin` app should be deployed. Getting data from the databases into Auth0 is more difficult compared to getting data into the app during deployments. This suggests a solution where a single source-of-truth ouside of the database is access by code in the `forms-deploy` repo both when Auth0 is configured by terraform and when the deployment pipeline is run.
+
+## Decision
+
+> what the team has decided to do
+
+Add configuration file(s) (YAML or JSON) to forms-deploy repo with list of domains and associated organisations
+    - Suggest folder `forms-deploy/config/data`
+
+Add Rake task to load domain configuration into database from file or environment variable
+    - See spike in forms-admin branch https://github.com/alphagov/forms-admin/compare/main...ldeb-spike-domains-read-into-database
+
+Change the pre-deploy configuration to inject the configuration for the environment from forms-deploy repo and then run the load domain configuration task
+    - See `forms-deploy/infra/modules/shared/scripts/db-migrations-pre-deploy.sh`, `forms-deploy/infra/modules/shared/scripts/db-migrations-ecs-run-task-input.jq`
+
+## Consequences
+
+> both positive and negative consequences of the decision
```

Comment:
Should we mention something here about the risk that the domains table might be updated without the corresponding change in the config? i.e. we'd have to ensure that the config remains the single source of truth.

Reply:
I've added something about this now.